tests/workflows/stream_try_node_annotation/tests/test_workflow.py

```diff
@@ -16,57 +16,103 @@ def test_workflow_stream__happy_path():
+    # assert workflow_initiated_events[1].parent is not None
+    # assert workflow_initiated_events[1].parent.type == "WORKFLOW_NODE"
+    # assert workflow_initiated_events[1].parent.node_definition == InnerNode
+    assert len(workflow_initiated_events) == 2
+
+    # node initiated events
+    node_initiated_events = [e for e in events if e.name == "node.execution.initiated"]
+    assert node_initiated_events[0].node_definition == InnerNode
+    assert node_initiated_events[0].model_dump(mode="json")["body"]["node_definition"] == {
         "name": "TryNode",
         "module": ["tests", "workflows", "stream_try_node_annotation", "workflow", "InnerNode", ADORNMENT_MODULE_NAME],
     }
-
-    assert events[2].name == "node.execution.streaming"
-    assert events[2].output.name == "processed"
-    assert events[2].output.is_initiated
-
-    assert events[3].name == "workflow.execution.streaming"
-    assert events[3].output.name == "final_value"
-    assert events[3].output.is_initiated
-
-    assert events[4].name == "node.execution.streaming"
-    assert events[4].output.delta == "apple apple"
-
-    assert events[5].name == "workflow.execution.streaming"
-    assert events[5].output.delta == "apple apple"
-
-    assert events[6].name == "node.execution.streaming"
-    assert events[6].output.name == "processed"
-    assert events[6].output.delta == "banana banana"
-
-    assert events[7].name == "workflow.execution.streaming"
-    assert events[7].output.delta == "banana banana"
-
-    assert events[8].name == "node.execution.streaming"
-    assert events[8].output.delta == "cherry cherry"
-
-    assert events[9].name == "workflow.execution.streaming"
-    assert events[9].output.delta == "cherry cherry"
-
-    assert events[10].name == "node.execution.streaming"
-    assert events[10].output.value == ["apple apple", "banana banana", "cherry cherry"]
-
-    assert events[11].name == "workflow.execution.streaming"
-    assert events[11].output.value == ["apple apple", "banana banana", "cherry cherry"]
-
-    assert events[12].name == "node.execution.fulfilled"
-    assert events[12].outputs.processed == ["apple apple", "banana banana", "cherry cherry"]
-
-    assert events[13].name == "workflow.execution.fulfilled"
-    assert events[13].outputs.final_value == ["apple apple", "banana banana", "cherry cherry"]
-
-    assert len(events) == 14
+    assert node_initiated_events[0].parent is not None
+    assert node_initiated_events[0].parent.type == "WORKFLOW"
+    assert node_initiated_events[0].parent.workflow_definition == StreamingTryExample
+    assert node_initiated_events[1].node_definition == WrappedNode
+    assert node_initiated_events[1].parent is not None
+    assert node_initiated_events[1].parent.type == "WORKFLOW"
+    assert node_initiated_events[1].parent.workflow_definition == InnerWorkflow
+    assert len(node_initiated_events) == 2
+
+    # inner node streaming events
```

Comment:
We can no longer guarantee order between all of these streaming events, so these filters make the assertions more deterministic